Mathlib/CategoryTheory/FullSubcategory.lean

```diff
@@ -0,0 +1,227 @@
+  inducedFunctor FullSubcategory.obj
+#align category_theory.full_subcategory_inclusion CategoryTheory.fullSubcategoryInclusion
+
+@[simp]
+theorem fullSubcategoryInclusion.obj {X} : (fullSubcategoryInclusion Z).obj X = X.obj :=
+  rfl
+#align category_theory.full_subcategory_inclusion.obj CategoryTheory.fullSubcategoryInclusion.obj
+
+@[simp]
+theorem fullSubcategoryInclusion.map {X Y} {f : X ⟶ Y} : (fullSubcategoryInclusion Z).map f = f :=
+  rfl
+#align category_theory.full_subcategory_inclusion.map CategoryTheory.fullSubcategoryInclusion.map
+
+instance FullSubcategory.full : Full (fullSubcategoryInclusion Z) :=
+  InducedCategory.full _
+#align category_theory.full_subcategory.full CategoryTheory.FullSubcategory.full
+
+instance FullSubcategory.faithful : Faithful (fullSubcategoryInclusion Z) :=
+  InducedCategory.faithful _
+#align category_theory.full_subcategory.faithful CategoryTheory.FullSubcategory.faithful
+
+variable {Z} {Z' : C → Prop}
+
+/-- An implication of predicates `Z → Z'` induces a functor between full subcategories. -/
+@[simps]
+def FullSubcategory.map (h : ∀ ⦃X⦄, Z X → Z' X) :
+    FullSubcategory Z ⥤ FullSubcategory
+        Z' where
+  obj X := ⟨X.1, h X.2⟩
+  map f := f
+#align category_theory.full_subcategory.map CategoryTheory.FullSubcategory.map
+
+instance (h : ∀ ⦃X⦄, Z X → Z' X) : Full (FullSubcategory.map h) where preimage f := f
+
+instance (h : ∀ ⦃X⦄, Z X → Z' X) : Faithful (FullSubcategory.map h) where
+
+@[simp]
+theorem FullSubcategory.map_inclusion (h : ∀ ⦃X⦄, Z X → Z' X) :
+    FullSubcategory.map h ⋙ fullSubcategoryInclusion Z' = fullSubcategoryInclusion Z :=
+  rfl
+#align category_theory.full_subcategory.map_inclusion CategoryTheory.FullSubcategory.map_inclusion
+
+section lift
+
+variable {D : Type u₂} [Category.{v₂} D] (P Q : D → Prop)
+
+/-- A functor which maps objects to objects satisfying a certain property induces a lift through
+    the full subcategory of objects satisfying that property. -/
+@[simps]
+def FullSubcategory.lift (F : C ⥤ D) (hF : ∀ X, P (F.obj X)) :
+    C ⥤ FullSubcategory P where
+  obj X := ⟨F.obj X, hF X⟩
+  map f := F.map f
+#align category_theory.full_subcategory.lift CategoryTheory.FullSubcategory.lift
+
+/-- Composing the lift of a functor through a full subcategory with the inclusion yields the
+    original functor. Unfortunately, this is not true by definition, so we only get a natural
+    isomorphism, but it is pointwise definitionally true, see
+    `FullSubcategory.inclusion_obj_lift_obj` and `FullSubcategory.inclusion_map_lift_map`. -/
+def FullSubcategory.liftCompInclusion (F : C ⥤ D) (hF : ∀ X, P (F.obj X)) :
+    FullSubcategory.lift P F hF ⋙ fullSubcategoryInclusion P ≅ F :=
+  NatIso.ofComponents (fun X => Iso.refl _) (by simp)
+#align
+  category_theory.full_subcategory.lift_comp_inclusion
+  CategoryTheory.FullSubcategory.liftCompInclusion
+
+@[simp]
+theorem FullSubcategory.inclusion_obj_lift_obj (F : C ⥤ D) (hF : ∀ X, P (F.obj X)) {X : C} :
```

Comment:
Shouldn't this name be
```suggestion
theorem fullSubcategoryInclusion_obj_lift_obj (F : C ⥤ D) (hF : ∀ X, P (F.obj X)) {X : C} :
```

Reply:
Maybe `fullSubcategoryInclusion` should itself be `FullSubcategory.inclusion`, to avoid some of these questions?